### DISCUSSION TITLE
Generalize Requestor Station Count to Account for Spoilage

### DISCUSSION THREAD

**Discussion by crackalackintv** [Opinionated-Blueprints/10-Books-Full-of-Rails#79] [Ideas]
Currently the Requestor stations count by the specific requested item in the boxes to set the limit. Given the nature of a requester, the items in there _should_ already be only the correct ones, but they can spoil. If I now have a full station of, for example, Bioflux, and it spoils away, a new train will subsequently arrive and block the station until all the spoilage has made its way out, as the buffer boxes are full.

I suggest just reading _Anything_ and divide it by it's stack size to get the filled number of slots and use that value for limits to correctly account for spoilage.

**Comment by Bocian-1** [comment]
Ah yes, I thought of items spoiling in trains but not in requester buffers. Good point

**Reply by Bocian-1** [reply]
I may need to modify it to a similar extent to the provider to also account for stack size differences between spoilage and things that spoil

**Reply by crackalackintv** [reply]
I gave it a shot, it's not cleaned up but should work. Let me know what you think, if you haven't solved it yourself already.

I take divide and mod by the stack size and add that up (full + half stacks), remove the buffer and compare that to the capacity of a train to set the limit.

```
0eNrdWu9u2zYQfxVCwL6kdGr9syUD2Ye2KPphwIolQFukhiHbtE1MljSKSpoVeYC9x55sT7I7SrJkW45Epc2WAoF9po6/Ox7vjrxTvhrzMGOJ4JE0Jl8Nvoij1JhcfzVSvo6CEMeiYMuMiSECHhr31ODRkn0xJuY9bWAKBJebLZN8MVjE2zmPAhmL2izrfkoNFkkuOcvlqB93syjbzpkAWFpCrYJUDtIk5FLCA2okcQqz4gilAdLAdp1zlxp3QI580MZYcsEWOYdpUWPFQ5xY6bZkuDgpMsU0SMIgihTyH1kAUgDIiGKxhfVQA5RPAqGUnxgXuIA4k0kmZ2CpWOTMIVtJA61wsARrfwlSBFGaxEIO5iyUjQuxq4UMz93DpTSIsB8jwuwkwtEWYemKcLVFmPuGasAcPQazm9pjWvMovmSi7uoNEkaVYTx/H99DT4ukiMPZnG2CGx4rhy1wZ/BsqbBSHK3/gsBZcZHKWRWB8i5BnW64kJny4ULJnGPwCzpxiiGw1Jy0Fws/G3RP8ixi8jYWvysNBVsak1UQpowaa8EYrBDCjR0KPp6DXLspCuAeE0Uec4f5qIvO3SSAAxmQBu7A1EuWLgRP8m0xrrlk2wvFPbiF7ZpeF9IGuZCLUtaUXJBt8IWAn/GIhHzL5econw1i2+fmPkXiFVllYUiWLOQ3TEBuJHLDU5LKADUiiyAiiYhvwCs+R78qq8DkrnI+R1fxmskNCLqF9AzQjMTqZ+FopDA0Oa3qSsRbcmQWEotDLSi53fDFhsEyCKwgjG+ZOG9Kk55uGNk/UBj989ffzzSQGrT6QQKpi5b/z1Dy6cP3r6ZoMnWjqYI+CCjNCPrUJ4LeqCtYwkRQONZLY3cl03Ho+6Y7gznsYb9haT3v+NLwfQ34oY8BLw8NeNbDgG/QgCci/c2EXMUyCCHEkmAB5sVYDIp4hj+IwAWkkSbzm48xv//U5v/Yx/xvD80/6GH+Tw+Y/1OT+XObo/UxT82z1YqJxh2oSpaUhWDIuO0wdkf76aPJ4PXlKk0gFf7JjMYItPVdwPX+swhkwWLTxwvKeXXL/NTDEUqcp7k/NG6Y85gN85/zhr18nht2UGzzKGWivbFyHFteE7h+1V0TceQNTpOIsYb+tq7+nga4pQvu74NnEVwT1yKG7y7tmsI8hYPljtfYdRo+omHTbQ8sU8NMpqaZLKuHmUx9M1UnjbqaDFIZJ83lZok87parFNwM4RKmkWKu8rwkZ2p+OlOlTxnX9THtGrXerMzhyt/dod7jcW0U1dOunsM6FrIylOzD6a5o+SNjKQ5BwZLKKRmQQ15zqhqrYW4u+ACPXavPuer7BkCfO743tH3LdOHLdEb2yLKHDuxBY7PPch5X+IyftPB537t1UD9+XvQ4fo4ldzxITlX1zbXteyzE50HKSOloXUv4q74l/CWDUruURmRMusp7cXINjVmjOj23bMmz7UBdkwV4WxKHrLlCqflZc6/aGj2m8Bk99SXqVR//ffct6s6PJwp3a/yYW+iTG/Dtt7qC9skBbx8oHoPlkgQQclglFjVjIlgK1m5sN1medjTU68VT0eBrNoSrnRx33cnv0REuN+jgNUnvtmuFl9yBolkkZ9gxnPEIrzRFaj7dceXYddkE4WqgdhIbh8Ve0qI3SUyyigVBOY3ba1c3SfXCNIikRjfgKKpOnKpp/jzdp8FQ5fvl8g3u4ehONbhniUEsWKc3uDiQ4Ttu9d66fBtN249fLXC7Bm7T9mNZC9wd1tAd2t4m7G0Xl7a3E7XAnbrqo1b013ro1nBYxx/T9pNMD58a2+DLrPxlOmPHs0fOuCbTo+0HoZ7FvHahPm0vLvR8wAdTnpI7bX3NU4UkXud+KyqCJcGHcBPUevOjccNsZv2ArItArGNyG6whuZzkvDzgJGkYS2D/cHZ5drSwixMgrxFEqg5svpAcsOzGnhT+Sq1IdWaJqpwgXePLJVTk5KR3SmPkrnR9dfauu7ZXlbal7ELRxgPB7Nc76noULLhYZFjgQsm6y/lF++tbngF4JGcpO5RxvN5aO1wyFuYlbdtqT1xrbLtf4+rZGs/p10t7tut1OzuL3e4so85gVjvYuF+77tnuhNfZeGa78XztGsfxW1GdYU1FOCbWGzlQ/1f5YAOyQ2vWMTWATR1gqzuw6+sA2xrA45PA4BvY6EFPux5Tk3rUnFJFWW5BWjBoFZRPHUU5MIYUfFLLVo99eGwO1agiTUXCiIm0WdIFbE6PanQuGVl3PEhDNVXQFvI7O3qc0xbKcpQGQNtIK34b5e7oigdpwK9or6Lh4Cpo+OXncx1cYb5sCzEth9pIO5We+FXqr+hCZxRT4gPtIu0ijZYsMb0av6LNGu2X9E63nPZL2kUaMW1lK1vxIyu18zXiELVzO+AXhaymaNQN0qXiGSGd64mPS7n2ThZ4ya4PnSel4uLMl+Bv8DExaj1tdQd6gMFUKQnvWjBc/fs0NW4AXfmlO7J8x/ddx7e8ERQ/9/8CjM06BA==
```

**Reply by Bocian-1** [reply]
Oh, I just uploaded a pre-release with my version of it :p
I took the chance and added mixed item requester support as well.
[v2.22.0](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/releases/tag/v2.22.0)

**Reply by crackalackintv** [reply]
Mixed item requests are going to be great for quality \o/

**Reply by crackalackintv** [reply]
Your implementation does not count partial stacks, so this could block a train after all, if I am not wrong here. That's why I added the `%` and if that is > 0, add a 1 to the stack size.

**Reply by Bocian-1** [reply]
Added check for partial stacks in [v2.22.1](https://github.com/Opinionated-Blueprints/10-Books-Full-of-Rails/releases/tag/v2.22.1)
